src/main/java/com/ryszardpanda/Library/controller/BookController.java

```diff
@@ -40,4 +41,30 @@ public Book updateBook(@PathVariable Integer id, @RequestBody Book updatedBook){
     public void deleteBookById(@PathVariable Integer id){
         boolean deletedBook = bookService.deleteBookById(id);
     }
+
+    @GetMapping("/books/search")
+    public List<Book> searchBooksByTitle(@RequestParam String title) {
+        return bookService.searchBooksByTitle(title);
+    }
+
+    @GetMapping("/books/searchByAuthor")
+    public List<Book> searchBookByAuthorAndYear(@RequestParam (required = false) String author, @RequestParam (required = false) Integer year) {
+        return bookService.searchBookByAuthorAndYear(author, year);
+    }
+
+    @PutMapping("books/updateRating")
+    public Book updateRating(@RequestParam Integer id, @RequestBody Book rating){
```

Comment:
@PathVariable instead of @RequestParam I think

Reply:
imo może zostać